File: SkinetV2.Contracts/Common/Pagination.cs
namespace SkinetV2.Contracts.Common
{
    public class Pagination<T>
        where T : class
    {
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public List<T> Data { get; set; }
        public Pagination(int pageIndex, int pageSize, List<T> data)
        {
            PageIndex = pageIndex;
            PageSize = pageSize;
            Data = data;
        }
    }
}

File: client/src/app/shared/models/common/pagination.ts
export interface Pagination<T> {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: Array<T>
}

export class Pagination<T> implements Pagination<T> {
    pageIndex: number;
    pageSize: number;
    count: number;
    data: T[] = [];
}